proxy/proxy.go

```diff
@@ -1127,6 +1127,12 @@ func (p *Proxy) do(ctx *context) (err error) {
 		}
 
 		backendStart := time.Now()
+		if d, ok := ctx.StateBag()[filters.ReadTimeout].(time.Duration); ok {
+			e := ctx.ResponseController().SetReadDeadline(backendStart.Add(d))
```

Comment:
Why not use conventional `err` name?

Reply:
Because we don't want to override the response error

Reply:
But it will be if-block scoped variable

Reply:
Oh, oh no it depends and `vet` will not allow this. :)